src/js/preloadWebComponents.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+var _ = require('underscore');
+
+function doesUserAgentContainString(str) {
+   return _.isString(window.navigator.userAgent) && window.navigator.userAgent.indexOf(str) >= 0;
+}
+
+module.exports = function() {
+   var needsWebComponents = !document.registerElement,
+       iosChrome = doesUserAgentContainString('CriOS'),
+       androidChrome;
+
+   androidChrome = doesUserAgentContainString('Android')
+      && doesUserAgentContainString('Chrome/')
+      && window.navigator.presentation;
+
+   // These check are based on the checks found in `cast_sender.js` which
+   // determine if `cast_framework.js` needs to be loaded
+   if ((androidChrome || iosChrome) && needsWebComponents) {
+      require('webcomponents.js'); // eslint-disable-line global-require
```

Comment:
Let's add a little bit more commenting here. These two things come to mind:

 1. See `https://github.com/silvermine/videojs-chromecast/issues/17`
 2. We use webcomponents.js@0.7.24 because that's what was used by the Chromecast framework at the time this was added.